app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -362,6 +362,29 @@ def enabled_dns_support?(vpc)
     vpc.describe_attribute(:attribute => 'enableDnsSupport', :vpc_id => vpc.vpc_id).enable_dns_support.value
   end
 
+  def enabled_internet_gateways?(vpc)
+    igw_ids = vpc.internet_gateways.map(&:internet_gateway_id)
+
+    unless igw_ids.empty? # Gateway attached
+      _log.debug("Found a gateway [#{igw_ids.first}] on VPC [#{vpc.id}]")
+
+      # Make sure gateway has vaild route
+      route_tables = vpc.route_tables.select do |route_table|
+        route_table.routes.any? { |route| route.gateway_id == igw_ids.first }
+      end
+
+      unless route_tables.empty?
+        _log.debug("Found route tables #{route_tables} on the gateway [#{igw_ids.first}]")
+        subnets = route_tables.map { |rt| rt.associations.first.subnet_id }
```

Comment:
I guess here we only care if there is one subnet so this use of "first" is ok, right?

Reply:
Yes, the API only has one entry per association returned.